.github/actions/create-jira-ticket/action.yml

```diff
@@ -0,0 +1,146 @@
+name: Nightly/QG incident tracking
+description: Create Jira ticket on nightly build/veracode qg failure
+
+inputs:
+  jira_base_url:
+    required: true
+    description: jira base url
+  jira_user_email:
+    required: true
+    description: jira user email
+  jira_api_token:
+    required: true
+    description: jira api token
+  module_name:
+    required: true
+    description: module name
+  reason_for_ticket_creation:
+    required: true
+    description: reason for ticket creation ("nightly", "veracode")
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get ticket elements from context
+      id: get_context
+      run: |
+        # Change context elements (summary, parent epic, etc.) that is checked depending on the reason the ticket is created
+        if [[ "${{ inputs.reason_for_ticket_creation }}" == "veracode" ]]; then
+            parent_epic_id=83818
+            parent_epic_key="AT-268"
+            ticket_summary="PR-${{ github.event.pull_request.number }} incident on ${{ inputs.module_name }}"
+            ticket_labels="[\"Veracode\", \"Pipeline\"]"
+            run_url="N/A"
+        elif [[ "${{ inputs.reason_for_ticket_creation }}" == "nightly" ]]; then
+            parent_epic_id=206242
+            parent_epic_key="MON-151547"
+            ticket_summary="$(date '+%Y-%m-%d') ${{ inputs.module_name }}-${{ github.ref_name }} nightly build failure"
+            ticket_labels="[\"Nightly\", \"Pipeline\"]"
+            run_url=$(gh run ls --json name,url --workflow web.yml --branch ${{ github.ref_name }} --limit 20 | jq --arg branch "${{ github.ref_name }}" 'map(select(.name | contains($branch))) | .[0].url')
```

Comment:
shouldn t we filter on dispatch event only (just in case someone runs web.yml for the branch around 6am, wouldn t it report another run)